### PR TITLE
chore(RHTAPWATCH-1237): Reorganize workspace-manager test suites

### PR DESCRIPTION
**Pull request by hmariset**
Since functional tests BeforeSuite runs all the tests, separate package for unit and functional tests.

Jira-Url: https://issues.redhat.com/browse/RHTAPWATCH-1237

### CODE REVIEW

**Comment by Omeramsc** [comment]
I'm a bit confused about the change. 
I can see we have 2 BeforeSuites; are they not affecting each other? We indeed need to separate the functional test's `BeforeSuite` from the unit tests, but we don't need a BeforeSuite for the unit tests. at least for now.
I also don't see the `TestGetNamespacesWithAccess` and `GetWorkspacesWithAccess` tests here, not in the code and not in the CI. why were they removed? 

We should see:
1. a test suite for the functional tests (beforesuite + aftersuite)
2. tests for `TestGetNamespacesWithAccess`
3. tests for `GetWorkspacesWithAccess`
4. tests for `TestRunAccessCheck`
5. The namespace provisioning test suite Gal added

the tests suites should be separated from each other. meaning the functional tests won't affect the unit tests at all, etc.

File: cmd/main_test.go [hunk 1]
```diff
@@ -267,14 +93,14 @@ var _ = BeforeSuite(func() {
...
-	createRole(k8sClient, "test-tenant", "namespace-access", []string{"create", "list", "watch", "delete"})
-	createRole(k8sClient, "test-tenant-2", "namespace-access-2", []string{"create", "list", "watch", "delete"})
-	createRoleBinding(k8sClient, "namespace-access-user-binding", "test-tenant", user1, "namespace-access")
-	createRoleBinding(k8sClient, "namespace-access-user-binding-2", "test-tenant", user2, "namespace-access")
-	createRoleBinding(k8sClient, "namespace-access-user-binding-3", "test-tenant-2", user2, "namespace-access-2")
+	utils.CreateRole(k8sClient, "test-tenant", "namespace-access", []string{"create", "list", "watch", "delete"})
+	utils.CreateRole(k8sClient, "test-tenant-2", "namespace-access-2", []string{"create", "list", "watch", "delete"})
+	utils.CreateRoleBinding(k8sClient, "namespace-access-user-binding", "test-tenant", user1, "namespace-access")
+	utils.CreateRoleBinding(k8sClient, "namespace-access-user-binding-2", "test-tenant", user2, "namespace-access")
+	utils.CreateRoleBinding(k8sClient, "namespace-access-user-binding-3", "test-tenant-2", user2, "namespace-access-2")
```
Comment (Omeramsc) [inline cmd/main_test.go]: Can we import the different functions from utils and keep the code as-is instead of changing all usages?

Comment (hmariset) [reply cmd/main_test.go]: moved these functions back to main_test.go

**Comment by hmariset** [comment]
@Omeramsc @gbenhaim Please review the PR.

File: cmd/main_test.go [hunk 2]
```diff
@@ -491,94 +507,58 @@ var _ = DescribeTable("TestGetNamespacesWithAccess", func(allNamespaces []k8sapi
...
+			createRoleBinding(k8sClient, "ns-namespace-access-user-binding-6", "ns-test-tenant-6", "user@konflux.dev", "ns-namespace-access-6")
 		})
-		It("returns only namespaces test-tenant and test-tenant-2", func() {
+		It("only returns ns-test-tenant-6 namespace", func() {
+			actualNs, err = getNamespacesWithAccess(e, c, authCl, allNamespaces)
 			Expect(actualNs).To(Equal(expectedNs))
 			Expect(err).NotTo(HaveOccurred(), "Unexpected error testing GetNamespacesWithAccess")
 		})
 		AfterEach(func() {
-			deleteRoleBinding(k8sClient, "ns-test-tenant-1", "ns-namespace-access-user-binding-1")
```
Comment (Omeramsc) [inline cmd/main_test.go:573]: What surprising me is, didn't this all work when we just had the functional tests? I saw no issues with the removal...

File: cmd/main_test.go [hunk 3]
```diff
@@ -110,38 +111,6 @@ func createNamespace(k8sClient client.Client, name string) (k8sapi.Namespace, er
...
-	ns := &k8sapi.Namespace{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: nsName,
-		},
-	}
-	err := k8sClient.Delete(context.Background(), ns)
-	Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error deleting the namespace: %s: %v\n", nsName, err))
-}
-
 func performHTTPGetCall(url string, header HTTPheader) (*HTTPResponse, error) {
```
Comment (Omeramsc) [inline cmd/main_test.go:114]: Maybe we should add this to `utils` as well

**Review by Omeramsc — COMMENTED** [review]
It is quite impossible to give a good review for such a large PR that moves things as well as change them. 
either separate the changes or set up an overview call with me (or both)

File: cmd/main_test.go [hunk 4]
```diff
@@ -34,20 +35,20 @@ import (
...
 type HTTPheader struct {
 	name  string
 	value string
 }
 
-type NamespaceRoleBinding struct {
-	Namespace   string
-	Role        string
-	RoleBinding string
+type HTTPResponse struct {
```
Comment (Omeramsc) [inline cmd/main_test.go]: Any reason we changed the order here?

Comment (hmariset) [reply cmd/main_test.go]: No, I first placed these in different file and again moved here. So, the order changed will change it to the original order.

File: cmd/main_test.go (hunk 2 above)
Comment (hmariset) [reply cmd/main_test.go:573]: When I use the same name it was in terminating state and not getting deleted. When checked with @gbenhaim he mentioned we cannot delete the objects in [envtest environment](https://book.kubebuilder.io/reference/envtest#testing-considerations)

File: cmd/main_test.go (hunk 3 above)
Comment (hmariset) [reply cmd/main_test.go:114]: When I add to the utils having trouble with the HTTPheader struct.

**Review by Omeramsc — COMMENTED** [review]
LGTM

**Review by Omeramsc — APPROVED** [review]
LGTM

**Pull request merged**